app.py

```diff
@@ -185,17 +192,21 @@ def set_frame_range(self, engine, in_frame, out_frame):
 
         elif engine == "tk-nuke":
             import nuke
-
-            # unlock
-            locked = nuke.root()["lock_range"].value()
-            if locked:
-                nuke.root()["lock_range"].setValue(False)
+            root = nuke.root()
+            box = QtGui.QMessageBox()
+            box.setWindowTitle("Sync Frame Range..")
+            box.setText(('Update scene frame range     \n\n'
+                         'from: %s-%s\n\n'
+                         'to:   %s-%s\n\n'
+                         '?') % (prev_in, prev_out, in_frame, out_frame))
+            box.addButton('Update', QtGui.QMessageBox.ActionRole)
+            box.addButton('Cancel', QtGui.QMessageBox.ActionRole)
+            if box.exec_() == 1:
+                return
             # set values
-            nuke.root()["first_frame"].setValue(in_frame)
-            nuke.root()["last_frame"].setValue(out_frame)
-            # and lock again
-            if locked:
-                nuke.root()["lock_range"].setValue(True)
+            root['first_frame'].setValue(in_frame)
+            root['last_frame'].setValue(out_frame)
+            root['lock_range'].setValue(True)
```

Comment:
yes, requested by Patrick David, after consensus with his team.

I know this gives them less control and I explained him it was better to keep the possibility to lock or not but he disregarded that.